src/main/java/com/duckblade/osrs/toa/features/scabaras/overlay/ScabarasOverlay.java

```diff
@@ -140,13 +140,30 @@ private void renderLocalMatching(Graphics2D graphics, Map<LocalPoint, MatchingTi
 			{
 				OverlayUtil.renderPolygon(graphics, canvasTilePoly, color, new Color(0, 0, 0, Math.min(color.getAlpha(), 50)), new BasicStroke(2));
 			}
-			if (name)
+
+			if (!name && !number)
+			{
+				return;
+			}
+
+			String text;
+			if (name && number)
 			{
-				Rectangle tileB = canvasTilePoly.getBounds();
-				Rectangle txtB = graphics.getFontMetrics().getStringBounds(mt.getName(), graphics).getBounds();
-				Point p = new Point(tileB.x + tileB.width / 2 - txtB.width / 2, tileB.y + tileB.height / 2 + txtB.height / 2);
-				renderTextLocationAlpha(graphics, p, mt.getName(), color);
+				text = mt.getNumber() + " - " + mt.getName();
 			}
+			else if (name)
+			{
+				text = mt.getName();
+			}
+			else
+			{
+				text = String.valueOf(mt.getNumber());
+			}
+
+			Rectangle tileB = canvasTilePoly.getBounds();
+			Rectangle txtB = graphics.getFontMetrics().getStringBounds(text, graphics).getBounds();
+			Point p = new Point(tileB.x + tileB.width / 2 - txtB.width / 2, tileB.y + tileB.height / 2 + txtB.height / 2);
+			renderTextLocationAlpha(graphics, p, text, color);
```

Comment:
this text centering stuff was not accounting for the number case, so the numbers were rendering no longer aligned with the tiles themselves. pre-building the text helps, and then this is also flexible enough to support name + number simultaneously